### DISCUSSION TITLE
Specify the tool pose sampling function

### DISCUSSION THREAD

**Discussion by ragesh88** [tesseract-robotics/tesseract_planning#564] [General]
Hi,

I wanted to use Descrates lite through the tesseract interface. I wanted the user to be able to specify different tool pose sampling functions to be used by descrates. What would be a good way to do this?

**Comment by marip8** [comment]
The `DescartesDefaultPlanProfile` profile class has a member called `target_pose_sampler` that you can set to a custom sampling function (with a specific function signature) or one provided by Tesseract (e.g., sample z-axis, sample fixed, etc.).

To get Descartes to do custom sampling like this, you'll need to create this profile class, set the `target_pose_sampler` member to your sampling function, and then add this profile class to the profile dictionary used for planning.
